ApiDoctor.Console/Program.cs

```diff
@@ -2562,6 +2557,210 @@ private static HttpRequest PreProcessUrlForSnippetGeneration(HttpRequest request
+
+            // generate permissions document
+            var permissionsDocument = new PermissionsDocument();
+            if (docFiles.Any() && !options.BootstrappingOnly)
+            {
+                using var fileStream = new FileStream(options.PermissionsSourceFile, FileMode.Open);
+                permissionsDocument = PermissionsDocument.Load(fileStream);
+            }
+
+            foreach (var docFile in docFiles)
+            {
+                bool finishedParsing = false, isBootstrapped = false;
+                var originalFileContents = File.ReadAllLines(docFile.FullPath);
+                var parseStatus = PermissionsInsertionState.FindPermissionsHeader;
+                int foundPermissionTablesOrBlocks = 0, foundHttpRequestBlocks = 0;
+                int insertionStartLine = -1, insertionEndLine = -1, httpRequestStartLine = -1, httpRequestEndLine = -1;
+                for (var currentIndex = 0; currentIndex < originalFileContents.Length && !finishedParsing; currentIndex++)
+                {
+                    var currentLine = originalFileContents[currentIndex];
+                    switch (parseStatus)
+                    {
+                        case PermissionsInsertionState.FindPermissionsHeader:
+                            if (currentLine.Trim() == "## Permissions")
+                                parseStatus = PermissionsInsertionState.FindInsertionStartLine;
+                            break;
+                        case PermissionsInsertionState.FindInsertionStartLine:
+                            if (currentLine.Contains("[!INCLUDE [permissions-table](")) // bootstrapping already took place
+                            {
+                                foundPermissionTablesOrBlocks++;
+                                isBootstrapped = true;
+                                if (!options.BootstrappingOnly)
+                                {
+                                    // find the permissions block start line
+                                    for (var i = currentIndex; i > 0; i--)
+                                    {
+                                        if (originalFileContents[i].Contains("<!-- {") || originalFileContents[i].Contains("<!--{"))
+                                        {
+                                            insertionStartLine = i;
+                                            insertionEndLine = currentIndex; // [!INCLUDE [permissions-table]... is the end of the insertion block
+                                            parseStatus = PermissionsInsertionState.FindHttpRequestHeader;
+                                            break;
+                                        }
+                                    }
+                                }
+                            }
+                            else if (currentLine.Contains('|') && currentLine.Trim().Contains("Permission type")) // found the permissions table
```

Comment:
```suggestion
                            else if (currentLine.Contains('|') && currentLine.Trim().Contains("Permission type",StringComparison.OrdinalIgnoreCase)) // found the permissions table
```
Just incase, can we make the string comparison ignore case?